File: two-sum.cc
#include <unordered_map>
#include <vector>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) 
    {
        std::unordered_map<int,std::vector<int>> num_map;
        std::vector<int> res;

        for (int i = 0; i < nums.size(); i++)
        {
            num_map[nums[i]].push_back(i);
        }

        for (const auto &num_pair : num_map)
        {
            int diff = target - num_pair.first;
            if (diff == num_pair.first)
            {
                if (num_pair.second.size() > 1)
                {
                    res = num_pair.second;
                    break;
                }
            }
            else
            {
                if (num_map.count(diff))
                {
                    res = {num_pair.second.front(), num_map[diff].front()};
                    break;
                }
            }
        }

        return res;
    }
};

File: two-sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {x: ind for ind, x in enumerate(nums)}

        for ind, val in enumerate(nums):
            if (target - val == val and d[val] != ind):
                return [ind, d[val]]
            elif (target - val != val and (target - val) in d):
                return [ind, d[target - val]]





        